internal/pkg/cri/pods.go

```diff
@@ -213,14 +213,16 @@ func stopAndRemove(ctx context.Context, stopAction StopAction, client *Client, p
 		return err
 	}
 
-	if err = client.StopPodSandbox(ctx, pod.Id); err != nil {
-		if grpcstatus.Code(err) == codes.NotFound {
-			return nil
-		}
+	if pod.State == runtimeapi.PodSandboxState_SANDBOX_READY {
```

Comment:
what if a pod failed to start due to cni error, is it okay to go ahead?

Reply:
we just don't try to stop a pod if it's not ready, as if it's not ready, it got stopped already